### ISSUE TITLE
False positive: isset operations variables existence: goto

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#1487]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 7.2

PHPEA does not understand `goto`.

```php
function f() {
        foo:
        isset($foo) && $foo = 'baz';
        $foo = 'bar';
        goto foo;
}
```

![image](https://user-images.githubusercontent.com/470626/72258016-ecde9280-3604-11ea-931d-5ce7f5c9c38b.png)

**Comment by kalessil** [comment]
@Bilge, I have problems understanding why it's a false-positive. Would you please take me to your point?

**Comment by Bilge** [comment]
@kalessil It's the same as the `do...while` loop. `$foo` is not set during the first iteration but is set in the second.

**Comment by ranvis** [comment]
I've experienced the similar warning:
```php
function f()
{
    retry:
    if (!isset($retried)) { # [EA] '$retried' seems to be not defined in the scope.
                            # Inspection info: Checks whether the operator argument is defined
                            # at all (due to refactoring it can be not existing anymore).
        $retried = true;
        goto retry;
    }
    
    for (;;) {
        if (!isset($retried2)) {
            $retried2 = true;
            continue;
        }
        break;
    }
}
```

**Comment by Bilge** [comment]
Still an issue almost two years later.

**Comment by orklah** [comment]
Goto must be the single most difficult syntax to analyse statically. Not sure if there's anything to be done, except maybe disable every inspection as soon as we see a goto in a scope...

**Comment by Bilge** [comment]
@orklah That would be a crap solution, of course, but actually better than dealing with the false positives we have now.

**Comment by ea-inspections-team** [comment]
We'll make the needed updates shortly.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @Bilge and @orklah, @ranvis for joining the discussion!
We managed to find a way to fix cases with goto (loops were handled properly already). Fixed!
